WordPress/src/test/java/org/wordpress/android/ui/posts/editor/media/AddLocalMediaToPostUseCaseTest.kt

```diff
@@ -25,7 +25,7 @@ import org.wordpress.android.ui.posts.editor.media.OptimizeMediaUseCase.Optimize
 import org.wordpress.android.util.MediaUtilsWrapper
 
 @RunWith(MockitoJUnitRunner::class)
-@UseExperimental(InternalCoroutinesApi::class)
+@OptIn(InternalCoroutinesApi::class)
```

Comment:
@ParaskP7 Could you explain your thoughts on why these changes should happen in this PR? The reason `UseExperimental` was replaced with `OptIn` in this PR was [because `UseExperimental` is deprecated](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-use-experimental/). What you seem to be suggesting is an overhaul of how and where these annotations are used, which might be the way to go - I'd have to look into it in more detail, but does it really belong to the Kotlin version update PR?

I feel like I am getting some mixed messages from you in this PR review as you are suggesting the dagger version update, which didn't require any code changes to be made, to be done in a separate PR, but then you are also suggesting including annotation improvements as part of it.

Reply:
👋 @oguzkocer !

Initially, please that this comment is only about suggestions (💡) for you to consider, nothing that you need to do if you don't want, have time of feel that is part of this PR.

> Could you explain your thoughts on why these changes should happen in this PR? The reason UseExperimental was replaced with OptIn in this PR was because UseExperimental is deprecated.

I only started suggesting these changes because you started replacing `UseExperimental` with `OptIn` in this PR, something that was optional to do. Thus, I noted an alternative to this, hence my suggestions.

> What you seem to be suggesting is an overhaul of how and where these annotations are used, which might be the way to go - I'd have to look into it in more detail, but does it really belong to the Kotlin version update PR?

I wouldn't say it is an overhaul, but if you feel that this is too much for this PR, then I just recommend you keeping the `@OptIn(InternalCoroutinesApi::class)` on the `NoDelayCoroutineDispatcher` level, but replace all other usage of this opt-in with `@InternalCoroutinesApi`. Wdyt?

> I feel like I am getting some mixed messages from you in this PR review as you are suggesting the dagger version update, which didn't require any code changes to be made, to be done in a separate PR, but then you are also suggesting including annotation improvements as part of it.

Yes, I agree on that. I would also have done that change in a separate PR or if this PR was done properly, and I mean it terms of it being scheduled and not done out of necessity. Then, I would have included these optimizations in it too, but as you said, an overhaul, along with any other optimizations that comes with this version of Kotlin. Does that make sense?

Thus, I only suggested improvements on these changes, since these changes occurred in this PR anyway.

Reply:
PS: To keep things moving and since the above is just a suggestion (💡) please feel free to resolve this conversation and consider turning this to another ticket that can be then worked in isolation.